src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,92 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+
 public class WorkWithFile {
+    private static final String NEWLINE_CHARACTER = System.lineSeparator();
+    private static final String DATA_SEPARATOR_CHARACTER = ",";
+    private static final int DATA_COLUMNS = 2;
+    private static final int OPERATION_TYPE_INDEX = 0;
+    private static final int AMOUNT_INDEX = 1;
+    private static final String[] REPORT_DATA = new String[] {"supply", "buy", "result"};
+    private static final int SUPPLY_INDEX = 0;
+    private static final int BUY_INDEX = 1;
+    private static final int RESULT_INDEX = 2;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String report = createReport(fromFileName);
+        writeToFile(report, toFileName);
+    }
+
+    private void writeToFile(String report, String toFileName) {
+        File file = new File(toFileName);
+        try {
+            file.createNewFile();
+        } catch (IOException e) {
+            throw new RuntimeException("Can't create the file " + toFileName, e);
+        }
+
+        try {
+            Files.write(file.toPath(), report.getBytes());
+        } catch (IOException e) {
+            throw new RuntimeException("Can't write data to the file " + toFileName, e);
+        }
+    }
+
+    private String createReport(String fromFileName) {
```

Comment:
lets divide this method - currently it has 2 tasks
reads data from file, and creates a report. Lets divide it, read data and parse it then